datalad/auto.py

```diff
@@ -132,13 +151,22 @@ def _proxy_open_name_mode(self, origname, origfunc, *args, **kwargs):
                     filearg = "name" if PY2 else "file"
                     if filearg not in kwargs:
                         # so the name was missing etc, just proxy into original open call and let it puke
-                        lgr.debug("No name/file was given, avoiding proxying")
-                        raise _EarlyExit
+                        raise _EarlyExit("no name/file was given")
                     file = kwargs.get(filearg)
+
                 if isinstance(file, int):
-                    lgr.debug(
-                        "Skipping operation on %i, already a file descriptor", file)
-                    raise _EarlyExit
+                    raise _EarlyExit("already a file descriptor")
+
+                if self._paths_cache is not None:
+                    filefull = file if isabs(file) else os.path.abspath(file)
+                    if filefull in self._paths_cache:
+                        raise _EarlyExit("considered before")
+                    else:
+                        self._paths_cache.add(filefull)
+
+                # TODO: Windows-proof?
+                if '/.git/' in file:
+                    raise _EarlyExit("we ignore paths under .git/")
```

Comment:
I guess that would not achieve the desired effect on windows. Why not `os.path.sep`?